### DISCUSSION TITLE
변수가 초기화되지 않은 상태를 제외한 reference Error 발생하는 경우

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#37] [궁금증]
- 변수가 초기화되지 않은 상태를 제외한 reference Error 발생하는 경우

**Comment by publdaze** [comment]
- ReferenceError 객체는 현재 범위에서 존재하지 않거나 초기화되지 않은 변수를 참조했을 때 발생하는 에러를 나타냄
   - [ReferenceError: "x" is not defined](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Errors/Not_defined)
     - (스코프 체인 상에) 선언되지 않은 변수 참조
   - [ReferenceError: assignment to undeclared variable "x"](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Errors/Undeclared_var)
     - strict mode에서 선언되지 않은 변수에 값 할당
   - [ReferenceError: can't access lexical declaration`X' before initialization](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Errors/Cant_access_lexical_declaration_before_init)
     - 변수가 초기화 되기 전에 엑세스(TDZ에서 변수 접근)
   - ~~[ReferenceError: deprecated caller or arguments usage](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Errors/Deprecated_caller_or_arguments_usage)~~ 확인 불가
   - [ReferenceError: invalid assignment left-hand side](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Errors/Invalid_assignment_left-hand_side)
     - 할당 연산자와 비교 연산자 간의 불일치
   - [ReferenceError: reference to undefined property "x"](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Errors/Undefined_prop)
     - strict mode에서 존재하지 않는 객체 속성에 접근 시도 (직접 test 해보니 에러 발생 안 함..!)

**Comment by pipisebastian** [comment]
[ReferenceError: "x" is not defined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Not_defined)
선언되지 않는 변수 참조

[ReferenceError: assignment to undeclared variable "x"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Undeclared_var)
선언되지 않은 변수에 값 할당 (strict mode에서!!!)

[ReferenceError: can't access lexical declaration 'X' before initialization](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Cant_access_lexical_declaration_before_init)
let, const 키워드 변수에 대해, 초기화되기전 접근

[ReferenceError: deprecated caller or arguments usage](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Deprecated_caller_or_arguments_usage)
 strict mode에서 사용할 수 없는 caller 또는 arguments를 참조

[ReferenceError: reference to undefined property "x"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Undefined_prop)
객체에서 존재하지 않는 프로퍼티에 접근
